drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -4312,8 +4312,8 @@ RID RenderingDeviceVulkan::shader_create(const Vector<ShaderStageData> &p_stages
 		}
 
 		pipeline_layout_create_info.pSetLayouts = layouts.ptr();
+		VkPushConstantRange push_constant_range;
```

Comment:
Could you add a comment to document the need to keep this in this scope?
Since compilers don't flag this apart from crashing on RADV, that would be good to avoid potential regressions.

Reply:
Not sure what kind of comment you'd like to see, something direct like
```
VkPushConstantRange push_constant_range; // do not move
```
or something more explainy at the place where the scope becomes important
```
// push_constant_range is declared outside, so we can safely use pointers                                                                                 
// to it in pipeline_layout_create_info.                                                                                                                  
pipeline_layout_create_info.pPushConstantRanges = &push_constant_range;
```
or something entirely different?

Reply:
The latter seems good.

Reply:
Well something like this:
```
// Needs to be declared in this outer scope, otherwise it may not outlive its assignment
// to pipeline_layout_create_info.
VkPushConstantRange push_constant_range;
If (...) {
```
```
VkPushConstantRange push_constant_range;

Reply:
@akien-mga resolved with your suggestion.